### DISCUSSION TITLE
请教正则表达式相关问题

### DISCUSSION THREAD

**Discussion by Renmr488** [ShixiangWang/self-study#101] [Q&A]
bxb_in='白血病|髓系(白血症|白细胞病)'
bxb_out='白血病(待[查排定]|？|\\?|不(排除|除外))'
bxb<-(str_subset(data$'最终临床诊断（合）','bxb_in'))&!(str_subset(data$'最终临床诊断（合）','bxb_out')) #R这一步报错，显示operations are possible only for numeric, logical or complex types，不知道什么原因呢？

想实现和python同样的功能，python代码如下：
bxb_in = '白血病|髓系白血症'
bxb_out = '白血病(待[查排定]|？|\?|不(排除|除外))'
bxb = (data['最终临床诊断（合）'].str.contains(bxb_in, case = False)) & (~data['最终临床诊断（合）'].str.contains(bxb_out, case = False))

**Comment by ShixiangWang** [comment]
`'bxb_in'` 是变量不是字符串吧？具体还得根据数据看看

**Reply by Renmr488** [reply]
![image](https://user-images.githubusercontent.com/62462250/185351133-48ab4669-8e2c-4f96-ab3d-d9266f1bd1fd.png)
是字符串

**Reply by ShixiangWang** [reply]
能否提供少量数据和可重复报错代码？
